benchmarks/piazza/piazza.rs

```diff
@@ -0,0 +1,332 @@
+            .default_value("100")
+            .help("Number of classes to prepopulate the database with"))
+        .arg(Arg::with_name("nusers")
+            .short("u")
+            .long("users")
+            .value_name("N")
+            .default_value("100")
+            .help("Number of users to prepopulate the database with"))
+        .arg(Arg::with_name("nposts")
+            .short("p")
+            .long("posts")
+            .value_name("N")
+            .default_value("10000")
+            .help("Number of posts to prepopulate the database with"))
+        .arg(Arg::with_name("csv")
+            .long("csv")
+            .required(false)
+            .help("Print output in CSV format."))
+        .arg(Arg::with_name("fanout")
+            .long("fanout")
+            .short("f")
+            .possible_values(&["all", "few"])
+            .takes_value(true)
+            .default_value("all")
+            .help("Size of the class fanout for each user"))
+        .arg(Arg::with_name("domain_config")
+            .long("dcfg")
+            .possible_values(&["single", "peruser"])
+            .takes_value(true)
+            .default_value("single")
+            .help("Domain assignment configuration"))
+        .arg(Arg::with_name("benchmark")
+            .possible_values(&["write", "migration"])
+            .takes_value(true)
+            .required(true)
+            .help("Benchmark configuration"))
+        .get_matches();
+
+
+    println!("Creating app...");
+    let mut app = Piazza::new();
+    println!("Done with app creation.");
+
+    let nusers = value_t_or_exit!(args, "nusers", i64);
+    let nclasses = value_t_or_exit!(args, "nclasses", i64);
+    let nposts = value_t_or_exit!(args, "nposts", i64);
+    let benchmark = args.value_of("benchmark").unwrap();
+    let domain_config = args.value_of("domain_config").unwrap();
+    let fanout = args.value_of("fanout").unwrap();
+    let csv = args.is_present("csv");
+
+    let class_putter = app.soup.get_mutator(app.class);
+    let user_putter = app.soup.get_mutator(app.user);
+    let taking_putter = app.soup.get_mutator(app.taking);
+    let post_putter = app.soup.get_mutator(app.post);
+
+    println!("Seeding...", );
+    populate_users(nusers, user_putter);
+    populate_classes(nclasses, class_putter);
+    populate_taking(nclasses, nusers, taking_putter, fanout);
+
+    match benchmark.as_ref() {
+        "migration" => {
+            for pid in 0..nposts {
+                post_putter.put(vec![
+                    pid.into(),
+                    (pid % nclasses).into(),
+                    (pid % nusers).into(),
+                    "post".into()
+                    ]);
+            }
+        },
+        "write" => {
+
+        }
+        _ => {
+            println!("wrong benchmark!");
+            return
+        }
+    }
+
+    println!("Finished seeding! Sleeping for 1 second...");
+    thread::sleep(time::Duration::from_millis(1000));
```

Comment:
I think we could probably reduce this to, say, 100ms. Should be plenty of time to drain the queues.